### PR TITLE
new homework04

### PR DESCRIPTION
**Pull request by vacilyok**
Please review my code!

### CODE REVIEW

File: homework04/pull_request.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        return
+    print('\n\n***************************************************************')
+    print('\tActive users:')
+    print('***************************************************************')
+    for user in sorted_user:
+        if user[1] > 1:
+            print(f"\t{user[0]} - ", user[1], ' pull requests')
+            # print(f"\t{user[0]} - \t{user[1]} pull requests")
+
+
```
Comment (Kromelky) [inline homework04/pull_request.py:24]: Add param support and filter state from cmd args

File: homework04/pull_request.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        print(result)
+        count_request = len(result)
+        open_request = open_request + count_request
+        if count_request == 0:
+            break
+        for elem in result:
+            user = elem['user']['login']
+            users.append(user)
+        page = page + 1
+
```
Comment (Kromelky) [inline homework04/pull_request.py:55]: Add label support, and group output by labels

File: homework04/pull_request.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    users = []
+    while True:
+        r = requests.get(
+            f'https://api.github.com/repos/{git_user}/{git_repo}/pulls?per_page=100&page={page}&state=open')
+        result = r.json()
+        print(result)
+        count_request = len(result)
+        open_request = open_request + count_request
+        if count_request == 0:
+            break
```
Comment (Kromelky) [inline homework04/pull_request.py:37]: Replace with list comprehantions

Comment (Kromelky) [inline homework04/pull_request.py:37]: Add output message

File: homework04/pull_request.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        url = 'https://github.com/vacilyok/Andersen-course'
+        print(f"\n\nLink to repository: {url}")
+    gitHubdata = url.split('/')
+    git_repo = gitHubdata[-1]
+    git_user = gitHubdata[-2]
+    page = 1
+    open_request = 0
+    users = []
+    label_dict = {}
+    while True:
```
Comment (Kromelky) [inline homework04/pull_request.py:38]: Add Github token authentication

Comment (vacilyok) [reply homework04/pull_request.py:38]: > 
> 
> Add Github token authentication

for what ?

**Review by Kromelky — APPROVED** [review]

**Pull request merged**